app/src/main/java/com/github/epfl/meili/database/Database.kt

```diff
@@ -1,11 +1,13 @@
 package com.github.epfl.meili.database
 
 import java.util.*
+import kotlin.collections.HashSet
 
-abstract class Database<T>(private val path: String): Observable() {
-    abstract var values: Map<String, T>
 
-    abstract fun addElement(uid: String?, element: T?)
+abstract class Database<T>(private val key: String): Observable() {
+    abstract var elements: Map<String, T>
+
+    abstract fun addElement(uid: String, element: T?)
```

Comment:
@aman3014 really cool that you managed to make it work without the nullable for uid, is it possible to do the same for element?

Reply:
Unfortunately it's not very easy. I read that we would have to switch over to using MockK (K for Kotlin) to do that, which can be time consuming.

Reply:
Sounds good. Thanks!